tests/pytests/unit/test_auth.py

```diff
@@ -0,0 +1,909 @@
+    you should NOT patch check_minions!
+    """
+    master_acl_valid_load["kwargs"]["username"] = "test_user_mminion"
+    master_acl_valid_load["user"] = "test_user_mminion"
+    await master_acl_clear_funcs.publish(master_acl_valid_load)
+    assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+
+
+async def test_master_not_user_glob_all(master_acl_clear_funcs, master_acl_valid_load):
+    """
+    Test to ensure that we DO NOT access to a given
+    function to all users with publisher acl. ex:
+
+    '*':
+        my_minion:
+            - my_func
+
+    Yes, this seems like a bit of a no-op test but it's
+    here to document that this functionality
+    is NOT supported currently.
+
+    WARNING: Do not patch this wit
+    """
+    master_acl_valid_load["kwargs"]["username"] = "NOT_A_VALID_USERNAME"
+    master_acl_valid_load["user"] = "NOT_A_VALID_USERNAME"
+    master_acl_valid_load["fun"] = "test.ping"
+    await master_acl_clear_funcs.publish(master_acl_valid_load)
+    assert "error" in master_acl_clear_funcs.event.fire_event.mock_calls.pop(-1).args[0]
+    assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+
+
+@pytest.mark.skip_on_windows(reason="PAM eauth not available on Windows")
+async def test_master_minion_glob(master_acl_clear_funcs, master_acl_valid_load):
+    """
+    Test to ensure we can allow access to a given
+    function for a user to a subset of minions
+    selected by a glob. ex:
+
+    test_user:
+        'minion_glob*':
+          - glob_mod.glob_func
+
+    This test is a bit tricky, because ultimately the real functionality
+    lies in what's returned from check_minions, but this checks a limited
+    amount of logic on the way there as well. Note the inline patch.
+    """
+    requested_function = "foo.bar"
+    requested_tgt = "minion_glob1"
+    master_acl_valid_load["tgt"] = requested_tgt
+    master_acl_valid_load["fun"] = requested_function
+    _check_minions_return = {"minions": ["minion_glob1"], "missing": []}
+    with patch(
+        "salt.utils.minions.CkMinions.check_minions",
+        MagicMock(return_value=_check_minions_return),
+    ):  # Assume that there is a listening minion match
+        await master_acl_clear_funcs.publish(master_acl_valid_load)
+    assert (
+        master_acl_clear_funcs.event.fire_event.called is True
+    ), "Did not fire {} for minion tgt {}".format(requested_function, requested_tgt)
+    assert (
+        master_acl_clear_funcs.event.fire_event.call_args[0][0]["fun"]
+        == requested_function
+    ), "Did not fire {} for minion glob".format(requested_function)
+
+
+async def test_master_function_glob(master_acl_clear_funcs, master_acl_valid_load):
+    """
+    Test to ensure that we can allow access to a given
+    set of functions in an execution module as selected
+    by a glob. ex:
+
+    my_user:
+        my_minion:
+            'test.*'
+    """
+    # Unimplemented
```

Comment:
```suggestion
async def _test_master_function_glob(master_acl_clear_funcs, master_acl_valid_load):
    """
    Test to ensure that we can allow access to a given
    set of functions in an execution module as selected
    by a glob. ex:

    my_user:
        my_minion:
            'test.*'
    """
    # Unimplemented
```

Reply:
Then lets not even collect it

Reply:
I didn't notice this while migrating the tests over. I've removed it since it's not testing anything.